components/centraldashboard/app/stackdriver_metrics_service.ts

```diff
@@ -0,0 +1,147 @@
+      });
+    } catch (err) {
+      console.error(
+          'Unable to fetch Node CPU Utilization metrics:',
+          err.details ? err.details : err);
+    }
+    return response;
+  }
+
+
+  async getPodCpuUtilization(interval: Interval): Promise<TimeSeriesPoint[]> {
+    const response: TimeSeriesPoint[] = [];
+    const filter = `metric.type="${CONTAINER_CPU_UTILIZATION_METRIC_TYPE}"`;
+    try {
+      const timeSeries = await this.fetchTimeSeries(filter, interval);
+      timeSeries.forEach((ts) => {
+        ts.points.forEach((p) => {
+          // Override the inferred type from the Point's TimeInterval
+          // since the actual response type is a proto Timestamp object
+          const endTime =
+              (p.interval.endTime as unknown) as monitoring.Timestamp;
+          response.push({
+            timestamp: Number(endTime.seconds),
+            label: ts.resource.labels['container_name'] ||
+                ts.resource.labels['pod_id'],
+            value: p.value.doubleValue,
+          });
+        });
+      });
+    } catch (err) {
+      console.error(
+          'Unable to fetch Pod CPU Utilization metrics:',
+          err.details ? err.details : err);
+    }
+    return response;
+  }
+
+  async getPodMemoryUsage(interval: Interval): Promise<TimeSeriesPoint[]> {
+    const response: TimeSeriesPoint[] = [];
+    const filter = `metric.type="${CONTAINER_MEMORY_USED_METRIC_TYPE}"`;
+    try {
+      const timeSeries = await this.fetchTimeSeries(filter, interval);
+      timeSeries.forEach((ts) => {
+        ts.points.forEach((p) => {
+          // Override the inferred type from the Point's TimeInterval
+          // since the actual response type is a proto Timestamp object
+          const endTime =
+              (p.interval.endTime as unknown) as monitoring.Timestamp;
+          response.push({
+            timestamp: Number(endTime.seconds),
+            label: ts.resource.labels['container_name'] ||
+                ts.resource.labels['pod_id'],
+            value: Number(p.value.int64Value),
+          });
+        });
+      });
+    } catch (err) {
+      console.error(
+          'Unable to fetch Pod Memory Utilization metrics:',
+          err.details ? err.details : err);
+    }
+    return response;
+  }
+
+  private async fetchTimeSeries(filter: string, interval: Interval):
+      Promise<gapi.client.monitoring.TimeSeries[]> {
+    try {
+      const [data] = await this.client.listTimeSeries({
+        name: this.client.projectPath(this.projectId),
+        filter,
+        interval: this.getTimeSeriesInterval(interval)
+      });
+      return data;
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  private getTimeSeriesInterval(interval: Interval):
+      {startTime: monitoring.Timestamp, endTime: monitoring.Timestamp} {
+    const now = Math.floor(Date.now() / 1000);
+    let secondsToSubtract = 0;
+    switch (interval) {
+      case Interval.Last5m:
+        secondsToSubtract = 60 * 5;
```

Comment:
You could refactor this, to simply hold minutes to Substract
```suggestion
        minsToSubtract = 5;
```
And then multiply the `60` in the bottom

Reply:
I agree. Changed.